tests/rptest/utils/si_utils.py

```diff
@@ -99,18 +117,64 @@ def read_batch(self):
             data = self.stream.read(records_size)
             if len(data) < records_size:
                 return None
-            assert len(data) == records_size
+            assert len(
+                data
+            ) == records_size, f"data len is {len(data)} but the expected records size is {records_size}"
             return header
         return None
 
-    def __iter__(self):
+    def __iter__(self) -> Iterator[Header]:
         while True:
             it = self.read_batch()
             if it is None:
                 return
             yield it
 
 
+def make_segment_summary(ntpr: NTPR, reader: SegmentReader) -> SegmentSummary:
+    """Read/parse segment and produce the summary"""
+    epoch: int = 0
+    max_int = int(2**63 - 1)
+    min_int = -1 * (max_int - 1)
+    base_offset = max_int
+    last_offset = min_int
+    base_timestamp = max_int
+    last_timestamp = min_int
+    num_data_batches = 0
+    num_conf_batches = 0
+    num_data_records = 0
+    num_conf_records = 0
+    size_bytes = 0
+    for header in reader:
+        base_offset = min(base_offset, header.base_offset)
+        last_offset = max(last_offset,
+                          header.base_offset + header.record_count - 1)
+        base_timestamp = min(base_timestamp, header.first_ts)
+        last_timestamp = max(last_timestamp, header.max_ts)
+        epoch = header.producer_epoch
+        if header.type == 19 or header.type == 2:
```

Comment:
Should this include `version_fence` batches? I'm not sure if they're written to user topics.
Also, would be nice to have an enum for this or at least a comment